### PR TITLE
Upgrade realm-core to v12.12.0

### PR DESCRIPTION
**Pull request by nielsenko**
- Update realm-core to v12.12.0
- Core no longer likes null pointer arguments on atlas function calls

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,12 +8,15 @@
 ### Fixed
 * Allow backlinks between files. ([#1015](https://github.com/realm/realm-dart/issues/1015))
 * Fix issue with accessing properties after traversing a backlink. ([#1018](https://github.com/realm/realm-dart/issues/1018))
+* Fix a race condition which could result in "operation cancelled" errors being delivered to async open callbacks rather than the actual sync error which caused things to fail (Core upgrade, since the introduction of async open).
```
Comment (nirinchev) [inline CHANGELOG.md:11]: We're not using async open in the dart SDK.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,12 +8,15 @@
...
+* Bootstraps will not be applied in a single write transaction - they will be applied 1MB of changesets at a time, or as configured by the SDK (Core upgrade, since v12.0.0).
```
Comment (nirinchev) [inline CHANGELOG.md:12]: The since here should probably be removed as we don't publicly advertise the Core version.

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,12 +8,15 @@
...
+* Fix database corruption and encryption issues on apple platforms, reported in several bugs listed in the PR. (Core upgrade, since v11.8.0)
```
Comment (nirinchev) [inline CHANGELOG.md:13]: Need to reword this or include a link to the PR. Also, need to remove "since ...".

File: lib/src/user.dart [hunk 4]
```diff
@@ -317,9 +317,9 @@ class FunctionsClient {
   /// Calls a remote function with the supplied arguments.
   /// @name The name of the Atlas function to call.
   /// @functionArgs - Arguments that will be sent to the Atlas function. They have to be json serializable values.
-  Future<dynamic> call(String name, [List<Object?>? functionArgs]) async {
+  Future<dynamic> call(String name, [List<Object?> functionArgs = const []]) async {
```
Comment (nirinchev) [inline lib/src/user.dart:320]: This is technically a breaking change - I'm not opposed to it, but we should at least mention it.

**Review by nirinchev — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (nielsenko) [reply CHANGELOG.md:13]: What do you prefer? Last time I as asked to remove the PR links

File: lib/src/user.dart (hunk 4 above)
Comment (nielsenko) [reply lib/src/user.dart:320]: Added to change log

File: CHANGELOG.md (hunk 3 above)
Comment (nirinchev) [reply CHANGELOG.md]: Generally, we don't include PR links, but the wording here mentions "issues linked in the PR". We should either explain what the issues are or link to the PR to allow people to read up themselves.

File: CHANGELOG.md (hunk 1 above)
Comment (nielsenko) [reply CHANGELOG.md]: Removed

File: CHANGELOG.md (hunk 3 above)
Comment (nielsenko) [reply CHANGELOG.md]: I have pushed a suggestion

**Review by nirinchev — APPROVED** [review]

**Pull request merged**